### ISSUE TITLE
Field filter on Postgres native question doesn't always work

### ISSUE THREAD

**Issue by dacort** [metabase/metabase#11883]
**Describe the bug**
When creating a field filter, sometimes I get a type error depending on how the question is loaded.

**Logs**

```
02-10 18:39:31 WARN middleware.process-userland-query :: Query failure {:status :failed,
 :class org.postgresql.util.PSQLException,
 :error "ERROR: operator does not exist: integer = character varying\n  Hint: No operator matches the given name and argument types. You might need to add explicit type casts.\n  Position: 169",
 :stacktrace
 ("org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2497)"
  "org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2233)"
  "org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:310)"
  "org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:446)"
  "org.postgresql.jdbc.PgStatement.execute(PgStatement.java:370)"
  "org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:149)"
  "org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:108)"
  "com.mchange.v2.c3p0.impl.NewProxyPreparedStatement.executeQuery(NewProxyPreparedStatement.java:431)"
  "clojure.java.jdbc$execute_query_with_params.invokeStatic(jdbc.clj:1072)"
  "clojure.java.jdbc$execute_query_with_params.invoke(jdbc.clj:1066)"
  "clojure.java.jdbc$db_query_with_resultset_STAR_.invokeStatic(jdbc.clj:1081)"
  "clojure.java.jdbc$db_query_with_resultset_STAR_.invoke(jdbc.clj:1075)"
  "clojure.java.jdbc$query.invokeStatic(jdbc.clj:1164)"
  "clojure.java.jdbc$query.invoke(jdbc.clj:1126)"
  "--> driver.sql_jdbc.execute$cancelable_run_query$fn__72979.invoke(execute.clj:225)"
  "driver.sql_jdbc.execute$do_with_ensured_connection.invokeStatic(execute.clj:202)"
  "driver.sql_jdbc.execute$do_with_ensured_connection.invoke(execute.clj:200)"
  "driver.sql_jdbc.execute$cancelable_run_query.invokeStatic(execute.clj:217)"
  "driver.sql_jdbc.execute$cancelable_run_query.invoke(execute.clj:213)"
  "driver.sql_jdbc.execute$run_query.invokeStatic(execute.clj:239)"
  "driver.sql_jdbc.execute$run_query.invoke(execute.clj:235)"
  "driver.sql_jdbc.execute$do_with_auto_commit_disabled.invokeStatic(execute.clj:286)"
  "driver.sql_jdbc.execute$do_with_auto_commit_disabled.invoke(execute.clj:274)"
  "driver.sql_jdbc.execute$do_in_transaction$fn__73000.invoke(execute.clj:291)"
  "driver.sql_jdbc.execute$do_in_transaction.invokeStatic(execute.clj:290)"
  "driver.sql_jdbc.execute$do_in_transaction.invoke(execute.clj:289)"
  "driver.sql_jdbc.execute$run_query_without_timezone.invokeStatic(execute.clj:314)"
  "driver.sql_jdbc.execute$run_query_without_timezone.invoke(execute.clj:313)"
  "driver.sql_jdbc.execute$execute_query$fn__73023.invoke(execute.clj:352)"
  "driver.sql_jdbc.execute$do_with_try_catch.invokeStatic(execute.clj:266)"
  "driver.sql_jdbc.execute$do_with_try_catch.invoke(execute.clj:260)"
  "driver.sql_jdbc.execute$execute_query.invokeStatic(execute.clj:346)"
  "driver.sql_jdbc.execute$execute_query.invoke(execute.clj:339)"
  "driver.sql_jdbc$fn__74193.invokeStatic(sql_jdbc.clj:44)"
  "driver.sql_jdbc$fn__74193.invoke(sql_jdbc.clj:43)"
  "query_processor$fn__48015$execute_query__48020$fn__48021.invoke(query_processor.clj:73)"
  "query_processor$fn__48015$execute_query__48020.invoke(query_processor.clj:68)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__40356.invoke(mbql_to_native.clj:40)"
  "query_processor.middleware.annotate$result_rows_maps__GT_vectors$fn__42359.invoke(annotate.clj:541)"
  "query_processor.middleware.annotate$add_column_info$fn__42265.invoke(annotate.clj:485)"
  "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__43300.invoke(cumulative_aggregations.clj:57)"
  "query_processor.middleware.resolve_joins$resolve_joins$fn__45110.invoke(resolve_joins.clj:184)"
  "query_processor.middleware.limit$limit$fn__43883.invoke(limit.clj:19)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__47875.invoke(results_metadata.clj:87)"
  "query_processor.middleware.format_rows$format_rows$fn__43719.invoke(format_rows.clj:76)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__40766.invoke(add_dimension_projections.clj:232)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__41417.invoke(add_source_metadata.clj:107)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__45160.invoke(resolve_source_table.clj:46)"
  "query_processor.middleware.add_row_count_and_status$add_row_count_and_status$fn__41254.invoke(add_row_count_and_status.clj:16)"
  "query_processor.middleware.driver_specific$process_query_in_context$fn__43375.invoke(driver_specific.clj:12)"
  "query_processor.middleware.resolve_driver$resolve_driver$fn__44774.invoke(resolve_driver.clj:22)"
  "query_processor.middleware.store$initialize_store$fn__47900$fn__47901.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:46)"
  "query_processor.store$do_with_store.invoke(store.clj:40)"
  "query_processor.middleware.store$initialize_store$fn__47900.invoke(store.clj:10)"
  "query_processor.middleware.async$async__GT_sync$fn__40271.invoke(async.clj:23)"
  "query_processor.middleware.async_wait$runnable$fn__42416.invoke(async_wait.clj:89)"),
 :query
 {:constraints {:max-results 10000, :max-results-bare-rows 2000},
  :type :native,
  :middleware {:userland-query? true},
  :native
  {:query "SELECT * FROM ORDERS WHERE {{order_id}}",
   :template-tags
   {"order_id" {:widget-type :id, :default "1", :name "order_id", :type :dimension, :id "37212e4c-c732-485b-ad97-03ff4a2ff8da", :dimension [:field-id 544], :display-name "Order", :required true}}},
  :info
  {:executed-by 1,
   :context :question,
   :card-id 33,
   :dashboard-id nil,
   :query-hash [-15, -79, 39, 21, 67, 39, 63, -70, 20, 106, -114, 120, -27, -95, 82, 62, -8, 116, 96, 32, -34, -1, -114, -65, -37, 73, 22, 40, 95, 84, -39, 55]},
  :parameters [{:type "id", :target ["dimension" ["template-tag" "order_id"]], :value ["1" "2"]}],
  :async? true,
  :cache-ttl nil}}
```

**To Reproduce**
1. On a postgres database, create a new native question
    SELECT * FROM ORDERS WHERE {{order_id}}
2. Configure "order_id" to be a Field filter on the ID column with a default value
3. Save the question and load the new question page
4. Add another ID in the filter dropdown
5. Notice we receive an error:
    ERROR: operator does not exist: integer = character varying Hint: No operator matches the given name and argument types. You might need to add explicit type casts. Position: 169
6. Clear the filter and run the query - notice that it runs fine with the default
7. Add 2 IDs to the filter box - notice that the query still runs fine
8. Reload the page now that `order_id=1&order_id=2` are in the URL - notice that we get the same error as above

**Expected behavior**
The filters work regardless of how they're added to the page.

**Screenshots**
n/a

**Information about your Metabase Installation:**
```json
{
  "browser-info": {
    "language": "en-US",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.87 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.4+11",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.4",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.4+11",
    "os.name": "Linux",
    "os.version": "4.19.76-linuxkit",
    "user.language": "en",
    "user.timezone": "Etc/UTC"
  },
  "metabase-info": {
    "databases": [
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "date": "2020-01-16",
      "tag": "v1.34.1",
      "branch": "enterprise-release-1.34.x",
      "hash": "a6df4f6"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
Annoying - I can work around it, but it should work

**Additional context**
Discovered while reproducing #11882
Possible regression of #10103

**Comment by flamber** [comment]
@dacort Is this friends with #11870? Keep this, close the other and up the P?

**Comment by paulrosenzweig** [comment]
Fixed in #11937
